tests/auth_tests/test_management.py

```diff
@@ -1522,6 +1528,208 @@ def test_permission_with_proxy_content_type_created(self):
+        Permission.objects.create(
+            codename="change_newmodel",
+            name="Can change newmodel",
+            content_type=ct,
+        )
+        Permission.objects.create(
+            codename="change_oldmodel",
+            name="Can change oldmodel",
+            content_type=ct,
+        )
+
+        call_command(
+            "migrate",
+            "auth_tests",
+            database="default",
+            interactive=False,
+            verbosity=0,
+        )
+
+        self.assertTrue(
+            Permission.objects.filter(
+                codename="change_oldmodel",
+                name="Can change oldmodel",
+            ).exists()
+        )
+        self.assertEqual(
+            Permission.objects.filter(
+                codename="change_newmodel",
+                name="Can change newmodel",
+            ).count(),
+            1,
+        )
+
+    def test_multiple_permission_types(self):
+        """Tests that all permission types (add/change/delete/view) are renamed"""
+        ct = ContentType.objects.create(app_label="auth_tests", model="oldmodel")
+        for action in ["add", "change", "delete", "view"]:
+            Permission.objects.create(
+                codename=f"{action}_oldmodel",
+                name=f"Can {action} oldmodel",
+                content_type=ct,
+            )
+
+        call_command("migrate", "auth_tests", verbosity=0)
+
+        for action in ["add", "change", "delete", "view"]:
+            self.assertFalse(
+                Permission.objects.filter(codename=f"{action}_oldmodel").exists()
+            )
+            self.assertTrue(
+                Permission.objects.filter(codename=f"{action}_newmodel").exists()
+            )
+
+        call_command(
+            "migrate",
+            "auth_tests",
+            "zero",
+            database="default",
+            interactive=False,
+            verbosity=0,
+        )
+        for action in ["add", "change", "delete", "view"]:
+            self.assertTrue(
+                Permission.objects.filter(codename=f"{action}_oldmodel").exists()
+            )
+            self.assertFalse(
+                Permission.objects.filter(codename=f"{action}_newmodel").exists()
+            )
+
+    def test_case_sensitivity(self):
+        """Tests that renaming works with mixed-case model names"""
+        ct = ContentType.objects.create(app_label="auth_tests", model="OldModel")
+        Permission.objects.create(
+            codename="change_oldmodel",
+            name="Can change OldModel",
+            content_type=ct,
+        )
+
+        call_command("migrate", "auth_tests", verbosity=0)
+        self.assertTrue(Permission.objects.filter(codename="change_newmodel").exists())
+
+        call_command(
+            "migrate",
+            "auth_tests",
+            "zero",
+            database="default",
+            interactive=False,
+            verbosity=0,
+        )
+
```

Comment:
```suggestion
        self.assertFalse(Permission.objects.filter(codename="change_newmodel").exists())

```
When I add the following assertion to check this has reversed, this fails. Can you look into this?

Reply:
I looked into this and to be honest that test is not needed and it didnt pass the test because when creating a contenttype for this test I put the "model=" with camel case which wont ever happen in real app since django manages the names by itself and lowers all the names of the models to lower case automatically.